src/DSE.Open/Text/Json/Serialization/JsonBinarySerializer.cs

```diff
@@ -11,24 +11,25 @@ namespace DSE.Open.Text.Json.Serialization;
 /// <summary>
 /// Helpers for reading/writing Json serialized data to/from memory.
 /// </summary>
+[RequiresDynamicCode(WarningMessages.RequiresDynamicCode)]
+[RequiresUnreferencedCode(WarningMessages.RequiresUnreferencedCode)]
 public static class JsonBinarySerializer
 {
+    private static readonly JsonSerializerOptions s_jsonSerializerOptions = JsonSharedOptions.RelaxedJsonEscaping;
+
     [RequiresDynamicCode(WarningMessages.RequiresDynamicCode)]
     [RequiresUnreferencedCode(WarningMessages.RequiresUnreferencedCode)]
     public static ReadOnlyMemory<byte> SerializeToUtf8Json<T>(T value, JsonSerializerOptions? jsonSerializerOptions = null)
     {
-        jsonSerializerOptions ??= JsonSharedOptions.RelaxedJsonEscaping;
-        var bufferWriter = new ArrayBufferWriter<byte>();
-        using var jsonWriter = new Utf8JsonWriter(bufferWriter);
```

Comment:
The `encoder` needs to be set on `jsonWriter` to be used. We were not passing any options, so it was using defaults (even though options are passed to the `Serialize` method.